task-ballerina/scheduler.bal

```diff
@@ -14,98 +14,154 @@
-        if (taskListener is error) {
-            return error SchedulerError(taskListener.message());
-        } else {
-            self.taskListener = taskListener;
-        }
-    }
+import ballerina/jballerina.java;
+import ballerina/time;
 
-    # Attaches the provided `service` to the task.
-    #
-    # + serviceToAttach - Ballerina `service` object, which needs to be attached to the task
-    # + attachments - Set of optional parameters, which need to be passed inside the resources
-    # + return - A `task:SchedulerError` if the process failed due to any reason or else ()
-    public isolated function attach(service object {} serviceToAttach, any... attachments) returns SchedulerError? {
-        var result = attachExternal(self.taskListener, serviceToAttach, ...attachments);
-        if (result is ListenerError) {
-            string message = "Failed to attach the service to the scheduler";
-            return error SchedulerError(message, result);
-        }
-    }
+# Configure the worker pool configurations.
+#
+# + workerCount - Specifies the number of workers that are available for the concurrent execution of jobs.
+#                 It should be a positive integer. The recommendation is to set a value less than 10. Default sets to 5.
+# + waitingTimeInMillis - The number of milliseconds the scheduler will tolerate a trigger to pass
+#                         its next-fire-time before being considered as `ignored the trigger`.
+# + return - A `task:TaskError` if the process failed due to any reason or else ()
+public isolated function configureWorkerPool(int workerCount = 5, int waitingTimeInMillis = 5000) returns TaskError? {
+    return configureThread(workerCount, waitingTimeInMillis);
+}
 
-    # Detaches the provided `service` from the task.
-    #
-    # + attachedService - Ballerina `service` object, which needs to be detached from the task
-    # + return - A `task:SchedulerError` if the process failed due to any reason or else ()
-    public isolated function detach(service object {} attachedService) returns SchedulerError? {
-        var result = detachExternal(self.taskListener, attachedService);
-        if (result is ListenerError) {
-            string message = "Scheduler failed to detach the service";
-            return error SchedulerError(message, result);
-        }
-    }
+# Schedule the given `job` for the given time. Once scheduled it will return a jobid # which can be used to manage
+# the job.
+#
+# + triggerTime - can be a specific UTC Time to trigger only one time
+# + job - Ballerina job, which is to be executed during the trigger.
+# + return - A job id or else `task:TaskError` if the process failed due to any reason
+public isolated function scheduleOneTimeJob(Job job, time:Time triggerTime) returns JobId|TaskError|error? {
+   time:Time time = time:currentTime();
+   string timeZone = time.zone.id;
+   time:Time value = check time:toTimeZone(triggerTime, timeZone);
```

Comment:
Rather than using `check`, shall we handle the error and pass the `task:Error`? So we can remove `error` from the return signature. and why to we need `?` in return param.

Reply:
Why are we converting the UTC time to local time. Can't we schedule the task at Utc time?

Reply:
Updated with a new design.